modules/twinkleblock.js

```diff
@@ -125,24 +133,39 @@ Twinkle.block.fetchUserInfo = function twinkleblockFetchUserInfo(fn) {
 Twinkle.block.callback.saveFieldset = function twinkleblockCallbacksaveFieldset(fieldset) {
 	Twinkle.block[$(fieldset).prop('name')] = {};
 	$(fieldset).serializeArray().forEach(function(el) {
-		Twinkle.block[$(fieldset).prop('name')][el.name] = el.value;
+		var param = Twinkle.block[$(fieldset).prop('name')][el.name];
+		// namespaces and pages for partial blocks will show up for each value
+		// Could maybe do a map().get().join() but this seems fine
+		if (param) {
+			Twinkle.block[$(fieldset).prop('name')][el.name] += '|' + el.value;
+		} else {
+			Twinkle.block[$(fieldset).prop('name')][el.name] = el.value;
+		}
 	});
 };
 
 Twinkle.block.callback.change_action = function twinkleblockCallbackChangeAction(e) {
 	var field_preset, field_template_options, field_block_options, $form = $(e.target.form);
+	// Make ifs shorter
+	var blockBox = $form.find('[name=actiontype][value=block]').is(':checked');
+	var templateBox = $form.find('[name=actiontype][value=template]').is(':checked');
+	var partial = $form.find('[name=actiontype][value=partial]');
+	var partialBox = partial.is(':checked');
```

Comment:
These lengthy constructs (here and in other functions) can be avoided by giving the `name` to the individual quickForm checkboxes (rather than to the list) eg. see batchdelete. Once that is done, the variables are readily accessed as `form.block.checked`, `form.template.checked` etc.